tokio/src/macros/cfg.rs

```diff
@@ -320,3 +320,22 @@ macro_rules! cfg_uds {
         )*
     }
 }
+
+macro_rules! cfg_syscall {
+    ($($item:item)*) => {
+        $(
+            #[cfg(all(feature = "syscall", tokio_unstable))]
+            #[cfg_attr(docsrs, doc(cfg(all(feature = "syscall", tokio_unstable))))]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_not_syscall {
```

Comment:
I don't think we can **remove** fns with a feature flag enabled. Maybe we have them panic if the **current** syscall trait is not the default one? Or, we add a hook for conversion to the syscall trait.

Reply:
In the most recent changeset I've attempted to address this by providing impls which panic when there isn't an equivalent functionality under the `syscall` flag, for example when trying to construct a `UdpSocket` from a `std::net::UdpSocket`.

Reply:
Features must be additive, so simply adding a feature __cannot__ change behavior.